src/components/InputBox/index.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { SetStateAction, useState } from 'react';
+import EyeIcon from '../../../public/assets/icons/eye';
+import EyeCloseIcon from '../../../public/assets/icons/eyeClose';
+
+type Props = {
+    type: string;
+    name: string;
+    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    placeholder?: string;
+    required: boolean;
+    className?: string;
+};
+
+const InputBox = (props: Props) => {
+    const [showPassword, setshowPassword] = useState<SetStateAction<boolean>>(false);
+    const { type, name, onChange, placeholder, required, className } = props;
+
+    return (
+        <>
+            <label htmlFor={name} className="block mb-2 text-sm font-medium text-gray-900 dark:text-white">
+                {name}
+            </label>
+            <div className="relative container mx-auto">
+                <input
+                    type={type == 'password' ? (showPassword ? 'text' : 'password') : 'text'}
```

Comment:
- Instead of having multiple conditional statements here can we do it something like
```js
<input type={showPassword ? 'text' : type} /> 
```

And also set a default value to type in props: 
```js
const { type = "text", name, onChange, placeholder, required, className } = props;
```

Reply:
updated